src/logger.ts

```diff
@@ -1,75 +1,72 @@
-  logger: winston.Logger
+  #client: Client
+  #group: string
+  severityThreshold: number
 
-  constructor(type: string, level: string, filename?: string) {
-    this.type = type
-    this.level = this.translateLogLevel(level)
+  constructor(client: Client, group: string, level: LoggerLevel = "info") {
+    this.#client = client
+    this.#group = group
+    this.severityThreshold = severity(level)
 
-    let transport
+    if (this.severityThreshold == UNKNOWN_SEVERITY) {
+      this.#client.integrationLogger.warn(
+        `Logger level must be "trace", "debug", "info", "warn" or "error", ` +
+          `but "${level}" was given. Logger level set to "info".`
+      )
 
-    if (type == "file") {
-      transport = new winston.transports.File({ filename: filename })
-    } else {
-      transport = new winston.transports.Console()
+      this.severityThreshold = severity("info")
     }
-
-    const logFormat = printf(({ level, message, timestamp }) => {
-      if (type == "file") {
-        return `[${timestamp} (process) #${process.pid}][${level}] ${message}`
-      } else {
-        return `[${timestamp} (process) #${process.pid}][appsignal][${level}] ${message}`
-      }
-    })
-
-    this.logger = winston.createLogger({
-      format: combine(timestamp({ format: "YYYY-MM-DDTHH:mm:ss" }), logFormat),
-      level: this.level,
-      transports: [transport]
-    })
   }
 
-  public error(message: string) {
-    this.logger.error(message)
+  trace(message: string, attributes?: LoggerAttributes) {
+    this.log(severity("trace"), message, attributes)
   }
 
-  public warn(message: string) {
-    this.logger.warn(message)
+  debug(message: string, attributes?: LoggerAttributes) {
+    this.log(severity("debug"), message, attributes)
   }
 
-  public info(message: string) {
-    this.logger.info(message)
+  info(message: string, attributes?: LoggerAttributes) {
+    this.log(severity("info"), message, attributes)
   }
 
-  public debug(message: string) {
-    this.logger.debug(message)
+  warn(message: string, attributes?: LoggerAttributes) {
+    this.log(severity("warn"), message, attributes)
   }
 
-  public trace(message: string) {
-    this.logger.silly(message)
+  error(message: string, attributes?: LoggerAttributes) {
+    this.log(severity("error"), message, attributes)
   }
 
-  /**
-   * Translates our logLevel to the one supported by Winston
-   */
-  private translateLogLevel(level: string): string {
-    switch (level) {
-      case "error":
-        return "error"
-      case "warning":
-        return "warn"
-      case "info":
-        return "info"
-      case "debug":
-        return "debug"
-      case "trace":
-        return "silly"
-      default:
-        return "info"
+  private log(
+    severity: number,
+    message: string,
+    attributes: LoggerAttributes = {}
+  ) {
+    if (severity < this.severityThreshold) {
+      return
     }
+
+    this.#client.extension.log(this.#group, severity, `${message}`, attributes)
```

Comment:
```suggestion
    this.#client.extension.log(this.#group, severity, message, attributes)
```
Does the message need to be wrapped like this?

Reply:
This is meant as a bit of defensive programming. The message should already be a string, so this does nothing most of the time.

But if it's not a string, this performs [string coercion](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String#string_coercion) on it. This ensures that whatever is passed to the extension is definitely a string.

Actually, looking at those docs now, it would be preferable to use `String()` to avoid raising a TypeError if a symbol is passed.

Reply:
Resolving with https://github.com/appsignal/appsignal-nodejs/pull/790/commits/c151a11b183c95c7d2bbd5cd1f505409456f75c8.